### DISCUSSION TITLE
INotificationHandler

### DISCUSSION THREAD

**Discussion by kalyanthumaty** [LuckyPennySoftware/MediatR#786] [Q&A]
I have read 1 notification can have N handlers, its a one to many relationship. In such case, what is the order that the handlers gets executed? Can we have a control to set the order of execution among handlers? Also, If I want to carry the output of one handler into another handler, or executing one handler based on the end result of another handler, is that possible?

**Comment by eskaufel** [comment]
You should not use events/notifications this way.
The order depends on the implementation of the di-container used, and it can change when new versions are released.

**Comment by KyMback** [comment]
What about managing execution order in `NotificationHandlerWrapperImpl`? We can create our own implementation of this, but we can't use it because we create implementation via Activator + it's private method 
<img width="776" alt="image" src="https://user-images.githubusercontent.com/26519973/179703357-59dc0dc5-9cff-49d9-bf6e-ef16f75b70a2.png">

**Comment by kalyanthumaty** [comment]
> You should not use events/notifications this way. The order depends on the implementation of the di-container used, and it can change when new versions are released.

@eskaufel Thanks for your response but I did not quiet get you. Let me ask my questions in a different way by taking a use case below.

I have a customer who placed an order => OrderPlacedEvent : INotification
 
After the order is placed, I have to perform the following three actions

1.  Save the order details to DB => OrderPlacedSaveToDB : INotificationHandler<OrderPlacedEvent>
2. Send email to customer => OrderPlacedEmailNotification : INotificationHandler<OrderPlacedEvent>
3. Set the status to "OrderPlaced" => OrderPlacedSetStatus : INotificationHandler<OrderPlacedEvent>

Question #1: Can I use MediatR INotification like above. 
Question #2: If the answer is yes, then in what order are these handlers get invoked?
Question #3: If the Handler#1 fails to do its job then the subsequent handlers should not be invoked. Is that possible?

Please respond. Thanks
